### PR TITLE
bug fix validate MERGE SET property: issue: 2637

### PR DESCRIPTION
**Pull request by OfirMos**

### CODE REVIEW

File: src/ast/ast.h [hunk 1]
```diff
@@ -37,6 +37,9 @@ bool AST_ContainsErrors(const cypher_parse_result_t *result);
 // asserts if there are no errors!
 void AST_ReportErrors(const cypher_parse_result_t *result);
 
+// Validate SET property.
+AST_Validation Validate_SETProperty(const cypher_astnode_t *set_item);
```
Comment (AviAvni) [inline src/ast/ast.h:41]: seems that is is not needed here you can move this to ast_validations.c

File: tests/flow/test_entity_update.py [hunk 2]
```diff
@@ -144,6 +144,39 @@ def test12_fail_update_property_of_non_alias_entity(self):
         except ResponseError as e:
             self.env.assertContains("RedisGraph does not currently support non-alias references on the left-hand side of SET expressions", str(e))
 
+    # Fail update an entity property when left hand side is not alias
+    def test12_2_fail_update_property_of_non_alias_entity(self):
+        try:
+            graph.query("MERGE (n:N) ON CREATE SET n.a.b=3 RETURN n")
+            self.env.assertTrue(False)
+        except ResponseError as e:
+            self.env.assertContains("RedisGraph does not currently support non-alias references on the left-hand side of SET expressions", str(e))
```
Comment (LiorKogan) [inline tests/flow/test_entity_update.py:153]: This is not the expected error.
The query is invalid.

File: tests/flow/test_entity_update.py [hunk 3]
```diff
@@ -144,6 +144,39 @@ def test12_fail_update_property_of_non_alias_entity(self):
...
+
+    # Fail update an entity property when left hand side is not alias
+    def test12_3_fail_update_property_of_non_alias_entity(self):
+        try:
+            graph.query("MERGE (n:N) ON CREATE SET n = {v: 1}, n.a.b=3 RETURN n")
+            self.env.assertTrue(False)
+        except ResponseError as e:
+            self.env.assertContains("RedisGraph does not currently support non-alias references on the left-hand side of SET expressions", str(e))
```
Comment (LiorKogan) [inline tests/flow/test_entity_update.py:161]: This is not the expected error.
The query is invalid.

File: tests/flow/test_entity_update.py [hunk 4]
```diff
@@ -144,6 +144,39 @@ def test12_fail_update_property_of_non_alias_entity(self):
...
+
+
+    # Fail update an entity property when left hand side is not alias
+    def test12_3_fail_update_property_of_non_alias_entity(self):
+        try:
+            graph.query("MERGE (n:N) ON MATCH SET n.a.b=3 RETURN n")
+            self.env.assertTrue(False)
+        except ResponseError as e:
+            self.env.assertContains("RedisGraph does not currently support non-alias references on the left-hand side of SET expressions", str(e))
```
Comment (LiorKogan) [inline tests/flow/test_entity_update.py:170]: This is not the expected error.
The query is invalid.

File: tests/flow/test_entity_update.py [hunk 5]
```diff
@@ -144,6 +144,39 @@ def test12_fail_update_property_of_non_alias_entity(self):
...
+
+    # Fail update an entity property when left hand side is not alias
+    def test12_3_fail_update_property_of_non_alias_entity(self):
+        try:
+            graph.query("MERGE (n:N) ON MATCH SET n = {v: 1}, n.a.b=3 RETURN n")
+            self.env.assertTrue(False)
+        except ResponseError as e:
+            self.env.assertContains("RedisGraph does not currently support non-alias references on the left-hand side of SET expressions", str(e))
```
Comment (LiorKogan) [inline tests/flow/test_entity_update.py:178]: This is not the expected error.
The query is invalid.

File: tests/flow/test_entity_update.py [hunk 6]
```diff
@@ -144,6 +144,39 @@ def test12_fail_update_property_of_non_alias_entity(self):
         except ResponseError as e:
             self.env.assertContains("RedisGraph does not currently support non-alias references on the left-hand side of SET expressions", str(e))
 
+    # Fail update an entity property when left hand side is not alias
+    def test12_2_fail_update_property_of_non_alias_entity(self):
```
Comment (AviAvni) [inline tests/flow/test_entity_update.py:148]: this tests can be part of the above test no need to add new function

**Review by AviAvni — CHANGES_REQUESTED** [review]
great see comments

**Pull request merged**